content/WebAuthn/WebAuthn_Developer_Guide/Overview.adoc

```diff
@@ -1,55 +1,92 @@
+FIDO2 is also backwards-compatible with FIDO U2F authenticators
+
+* **Single factor:** Username + FIDO U2F credential
+* **Second factor:** Username + password + FIDO U2F credential
 
-The following sections in this guide describe the basic FIDO2/WebAuthn user flows:
+== FIDO2/WebAuthn application architecture
 
-* Registration Sequence
-* Authentication Sequence
+image::fido2_app_architecture.png[]
+
+In general, a FIDO2/WebAuthn authentication architecture involves a conversation between a computing environment controlled by a Relying Party and one controlled by the user to be authenticated.
 
+=== FIDO2 Authenticator
+The user environment may consist of a client computing device with internal FIDO2 authenticator. The user may also have an external FIDO2 authenticator, such as a YubiKey, which can roam between devices. The authenticator makes credentials, generates cryptographic proof of user authentication, and manages the PIN.
 
-=== Building a WebAuthn Experience
+=== Client/Platform
+The client is the bridge between the authenticator and the RP. It implements CTAP2 and client-side WebAuthn API. The client could be a browser exposing the WebAuthn API to web applications, or an OS subsystem exposing a platform-specific FIDO API to native applications such as mobile or desktop apps.
 
-The Yubico Developer Program provides workshops, documentation, implementation guides, APIs, and SDKs. To get started,
+=== Relying Party
+The Relying Party's environment consists conceptually of at least a web server and the server-side portions of a web application, plus a WebAuthn server. The WebAuthn server has a trust store, containing the (public) trust anchors for the attestation of FIDO2 Authenticators. Note: a trust store is needed only if RP cares about attestation metadata.
 
-1. Sign up to the Yubico Developer Program to access the resources for implementing the FIDO2 and Web Authentication specifications.
-2. Watch FIDO2 / Web Authentication webinars to learn more about the core concepts and how to integrate WebAuthn RP operations into an authentication server.
-4. Read the developer guide for an introduction to WebAuthn/FIDO2 using Yubico’s Java WebAuthn Server.
 
-link:WebAuthn_Client_Registration.html[Next: WebAuthn Client Registration]
+== Key Points
+* FIDO2 credentials are based on an public key cryptographic key pair.
+* The identity provider validates user identity and maps a FIDO2 credential public key to a user account during the registration step
+* FIDO2 credentials can be generated in hardware devices (e.g. security keys, mobile phones, laptops, etc…)  or software, based on policy.
+* Authentication uses a cryptographic key tied to a device and, optionally, something that the person knows (a PIN) or something that the person is (biometric gesture). PINs and biometric gestures do not roam between devices and are not shared with the server; they are stored locally on a device.
```

Comment:
It's often ambiguous whether "device" means "computer/phone" or "YubiKey", so let's use more precise terms where the distinction is significant.

```suggestion
* Authentication uses a cryptographic key tied to an authenticator and, optionally, something that the person knows (a PIN) or something that the person is (biometric gesture). PINs and biometric gestures do not roam between authenticators and are not shared with the server; they are stored locally on an authenticator.
```